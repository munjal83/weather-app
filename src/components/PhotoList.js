import React, { Component } from 'react';
import { photos } from "../assets/photos";
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';


const styles = {
    root: {
        width: '100%',
        maxWidth: 600,
        backgroundColor: "fff",
        margin: '0 auto',
        padding: 20,
    },
};

class PhotoList extends Component {
    constructor(props) {
        super(props);
        this.photoList = photos;
    }

    getPhotoList() {
        return this.photoList.map(photo => <div className="text-center"><a href={photo.src}><h3>{photo.name}</h3></a></div>);
    }

    render() {
        return (
            <div style={styles.root}>
                <Paper elevation={3}>
                    <List>
                        <Typography variant="display1" component="h3" align="center" gutterBottom>
                            Photo List
                        </Typography>
                        <ListItem>
                            <Typography variant="subheading" component="h4" align="left" gutterBottom>
                                {this.getPhotoList()}
                            </Typography>
                        </ListItem>
                    </List>
                </Paper>
            </div>
        );
    }
}

export default PhotoList;