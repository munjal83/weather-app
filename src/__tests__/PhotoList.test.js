import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import PhotoList from '../components/PhotoList';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<PhotoList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('PhotoList Component Snapshot',()=>{
  test('testing photolist component', () => {
  let tree = create(<PhotoList />)
  expect(tree.toJSON()).toMatchSnapshot();
})
});