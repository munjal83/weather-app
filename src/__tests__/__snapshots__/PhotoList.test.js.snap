// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoList Component Snapshot testing photolist component 1`] = `
<div
  style={
    Object {
      "backgroundColor": "fff",
      "margin": "0 auto",
      "maxWidth": 600,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation3-72 MuiPaper-rounded-68"
  >
    <ul
      className="MuiList-root-94 MuiList-padding-95"
    >
      <h3
        className="MuiTypography-root-98 MuiTypography-display1-102 MuiTypography-gutterBottom-115 MuiTypography-alignCenter-111"
      >
        Photo List
      </h3>
      <li
        className="MuiListItem-root-123 MuiListItem-default-126 MuiListItem-gutters-130"
        disabled={false}
      >
        <h4
          className="MuiTypography-root-98 MuiTypography-subheading-105 MuiTypography-gutterBottom-115 MuiTypography-alignLeft-110"
        >
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280282/CNreflection.jpg"
            >
              <h3>
                CN Tower Reflection
              </h3>
            </a>
          </div>
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280282/Biosphere.jpg"
            >
              <h3>
                Biosphere
              </h3>
            </a>
          </div>
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280282/TOpurple.jpg"
            >
              <h3>
                Purple Skyline Toronto
              </h3>
            </a>
          </div>
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280281/Tonpsq.jpg"
            >
              <h3>
                Nathan Phillip Square
              </h3>
            </a>
          </div>
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280281/Gooderham.jpg"
            >
              <h3>
                Gooderham Building
              </h3>
            </a>
          </div>
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280281/MontrealP.jpg"
            >
              <h3>
                Montreal Purple Skyline
              </h3>
            </a>
          </div>
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280281/ToW.jpg"
            >
              <h3>
                Toronto Winter
              </h3>
            </a>
          </div>
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280281/MoW.jpg"
            >
              <h3>
                Montreal Winter
              </h3>
            </a>
          </div>
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280281/MoS.jpg"
            >
              <h3>
                Montreal Skyline
              </h3>
            </a>
          </div>
          <div
            className="text-center"
          >
            <a
              href="https://res.cloudinary.com/asgard83/image/upload/v1565280281/Mobr.jpg"
            >
              <h3>
                Bridge
              </h3>
            </a>
          </div>
        </h4>
      </li>
    </ul>
  </div>
</div>
`;
